bpf/lib/nodeport.h

```diff
@@ -2535,8 +2548,10 @@ int tail_nodeport_nat_egress_ipv4(struct __ctx_buff *ctx)
 		goto drop_err;
 	}
 
-	ret = fib_redirect_v4(ctx, ETH_HLEN, ip4, true, &ext_err,
-			      ctx_get_ifindex(ctx), &oif);
```

Comment:
> tail_nodeport_nat_egress_ipv4() potentially adds outer tunnel headers

That's right, just 10 lines above we have a nodeport_add_tunnel_encap() followed by ctx_redirect().

> The call to fib_redirect_v4() then decrements outer header's TTL

I have a question for this. In the case of tunneling, we'll not reach here (fib_redirect_v4()) because the packet has been bpf_redirect()-ed to cilium_vxlan at 2529L, so the outer header has no chance to get TTL decremented in the `fib_redirect_v4()`; sure the inner ip4 doesn't update its TTL either, but do we need to update TTL for a bpf_redirect()?

Reply:
> I have a question for this. In the case of tunneling, we'll not reach here (`fib_redirect_v4()`) because the packet has been `bpf_redirect()`-ed to cilium_vxlan at 2529L, so the outer header has no chance to get TTL decremented in the `fib_redirect_v4()`

For XDP it doesn't get redirected to cilium_vxlan (there's no XDP TX / REDIRECT support in the vxlan driver). Instead ctx_set_encap_info() manually builds the headers and returns ifindex 0, so we do a FIB lookup based on the new outer headers.